llvmlite/ir/module.py

```diff
@@ -31,6 +31,17 @@ def add_metadata(self, operands):
             md = self._metadatacache[key]
         return md
 
+    def add_debug_info(self, kind, operands, is_distinct=False):
+        n = len(self.metadata)
+        operands = tuple((key, operands[key]) for key in operands)
```

Comment:
How about `sorted(operands.items())`? This will give a you a deterministic order (nicer for testing).